### DISCUSSION TITLE
Failing to adding a new mysql database connection in superset

### DISCUSSION THREAD

**Discussion by ghost** [apache/superset#26093] [Q&A / Help]
Hi All, 
Can anyone guide me and tell me what might be the issue? 
I have set up a superset instance by "virtualenv" method on a ubuntu server. I have also installed and set up mysql database on ubuntu server. I am not able to add database connection on superset even though mysql server is up and running. I get errors like "
![Screenshot (3)](https://github.com/apache/superset/assets/151936944/ea31f725-f3fc-4a58-9b6c-a7e61ca24ce4)

can anyone tell me, what should be the Alchemy URL format for mysql , currently the one I'm using is 
mysql://username:password@hostname:port/databasename

**Comment by MahendraRao** [comment]
@venkateshgdixit  The SQLAlchemy URL format you are using for MySQL in Superset seems correct, but there are a few things you should check to troubleshoot the "Connection failed, please check connection settings" error:

a) Verify MySQL Server Connection ==> Ensure that you can connect to your MySQL server using the provided credentials and that the server is accessible from the machine where Superset is running.
b) Check MySQL Server Configuration ==> Is it configured to allow external connections?  If Superset is not running on the same machine as the MySQL server. You may need to update the MySQL server's 'bind-address' in the 'my.cnf' configuration file.
c) Firewall Settings ==> typically 3306 for MySQL ==> Is it OPEN?
d) Verify MySQL User Privileges ==> Make sure that there are no typos or errors in the information provided
e) Install MySQL Connector Installation (recheck this installation one more time) 
f) Superset Configuration ==> Check your Superset configuration files (superset_config.py or any custom configuration file) to ensure that the database connection details are correctly specified. The configuration file typically contains a section like:
SQLALCHEMY_DATABASE_URI = 'mysql://username:password@hostname:port/databasename'

After checking and making necessary adjustments, restart your Superset server to apply the changes and attempt to add the database connection again. If the issue persists, reviewing the Superset logs may provide more detailed error messages that can help identify the problem.

**Reply by ghost** [reply]
Thank you @MahendraRao for your quick and on-point answer. I checked the points that you mentioned in your answer. Below is what happened
a) I was able to connect to MySQL Server via command prompt with the credentials and was accessible from the machine on which superset is running.
b) It is running on the same machine as the MySQL server.
c) I verified that 3306 is open by running the command " sudo netstat -tulpn | grep 3306 " and got the output of "LISTEN"
d)I am using the root user credentials and verified the information
e)mysql connector installed as expected. Did try uninstalling and re-installing.
f)I searched for the superset_config.py file but couldn't find it. So had to create the file and insert the contents via nano editor including SQLALCHEMY_DATABASE_URI = 'mysql://username:password@hostname:port/databasename and placed at path  /home/file/superset_config.py. 

With all these changes and after restart. I'm still getting the same error. Also I don't see any updated logs for these erros in settings->action log.  I'm mainly trying to get the functionality of csv upload after adding mysql . Suggest me if there are any easier ways to achieve it. Thanks in advance.
